### DISCUSSION TITLE
Omit a screen from history?

### DISCUSSION THREAD

**Discussion by samsnori** [react-navigation/react-navigation#11910] [Q&A]
When navigating to a screen is there a way that skips the screen from being added to the history?

I'm not talking about replacing the current screen with the new-screen (which removes the current screen from history), rather I want to navigate from screen A to screen B and make sure that screen B is not added to the history. 

_As an example, consider this situation:_

- I navigate to Screen A
- From A, I navigate to B (*I want to prevent this screen B from being added to the history*)
- Then I receive a notification (e.g. OS level notification) that navigates to Modal M (via deeplink)
- Now, pressing the "Back" button on this model M, should bring me back to A, not to B.

**Comment by satya164** [comment]
You can do it with your own custom navigator or by using a custom router with a built in navigator. For example, by default "routes" array is used for history, but in your custom router you can add a "history" array in the navigation state and only add items that should be in history to that.

**Reply by samsnori** [reply]
Thanks @satya164 I had a quick look at the documentation about creating [customer routers](https://reactnavigation.org/docs/custom-routers/#:~:text=You%20can%20make%20your%20own,with%20routeNames%20and%20routeParamList%20properties.) and [custom navigators](https://reactnavigation.org/docs/custom-routers/#:~:text=You%20can%20make%20your%20own,with%20routeNames%20and%20routeParamList%20properties.).  The documentation does mention something about [clearing the history ](https://reactnavigation.org/docs/custom-routers#custom-navigation-actions) which is probably a good starting point.
